src/main/java/org/cactoos/text/RegexText.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.text;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import org.cactoos.Text;
+
+/**
+ * Returns a valid regex expresion.
+ *
+ * @author Ix (ixmanuel@yahoo.com)
+ * @version $Id$
+ * @since 0.12
+ */
+public final class RegexText implements Text {
```

Comment:
Could we review it together? because I think that if we have a compareTo() why not a matches(). Although, the underlying concept in RegexText is to create a Type. It has the same meaning as StringAsText (TextOf sounds more like a type) because in Java there is not a real type for Regex, it relies on a string. 

Some refections:

1. RegextText was conceived to distinguish a simple text from a regex and also to verify its syntax. The difference with StringAsText is that the string is statically verified while RegexText not. In the proposal ReplacedAllText it uses ReplacedRegexText that also uses RegexText.
2. If we implement matches() in RegextText we could be tempted to add more methods like: Split, ReplacedFirst and ReplacedAll but, matches is reasonable, because we could say that it is its natural behaviour.
3. Once added matches() we have to create a new protocol and thus, RegexText is not more a  consistent decoration of Text.

These previous ideas are what they are, just ideas. I can't visualise a big issue and I think we could add it. So let me present you an implementation in order to be sure that we are understanding the same.

```java
    @Override
    public boolean matchesIn(final Text text) {
        return this.origin.matches(text.asString());
    }
```
with a protocol: Match?